src/Kernels.cpp

```diff
@@ -44,22 +43,25 @@ void InitKernel(int DoPlaces, double norm)
 		Kernel = PowerExpKernel;
 	else
 		ERR_CRITICAL_FMT("Unknown kernel type %d.\n", P.KernelType);
-#pragma omp parallel for private(i) schedule(static,500) //added private i
-	for (i = 0; i <= P.NKR; i++)
+
+#pragma omp parallel for schedule(static,500) default(none) \
+		shared(P, Kernel, nKernel, nKernelHR, norm)
+	for (int i = 0; i <= P.NKR; i++)
```

Comment:
This is a non-obvious change.  However, I am happy about it:
 * `i` is the loop variable so automatically private
 * `default(none)` ensures we explicitly label the data-sharing of all variables.
 * All other variables are shared as they should be.